tests/framework/extensions/users/users.go

```diff
@@ -0,0 +1,69 @@
+package users
+
+import (
+	"time"
+
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func CreateUserWithRole(rancherClient *rancher.Client, user *management.User, role string) (*management.User, error) {
```

Comment:
just a thought would this be more useful if role was a variadic parameter?

Reply:
Hmmm, how would that work in this function. We create multiple GlobalRoleBindings for a single user?

Reply:
yeah we would just loop through the roles and add each one

Reply:
If you think it's necessary. I can make the changes for sure

Reply:
updated